docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install

```diff
@@ -923,3 +927,111 @@ function su_humsci_profile_update_9707() {
+
+  if ($fields) {
+    foreach ($fields as $field) {
+      /** @var Drupal\field\Entity $field */
+
+      // Delete the field configuration (instance) from the paragraph type.
+      $field->delete();
+
+      // Load the corresponding field storage.
+      $field_storage = FieldStorageConfig::load($field->getName());
+      if ($field_storage) {
+        // Check if the field storage is used elsewhere.
+        if (!$field_storage->isLocked()) {
+          // Delete the field storage if it's no longer in use by any other bundles.
+          $field_storage->delete();
+        }
+      }
+    }
+  }
+
+  // Delete the paragraph itself.
+  $paragraph_type = ParagraphsType::load($paragraph_machine_name);
+  if ($paragraph_type) {
+    $paragraph_type->delete();
+  }
+
+  // Remove hs_webform from config ignored configs.
+  $config_storage = \Drupal::service('config.storage.sync');
+
+  $configs_to_change = [
+    'node.hs_basic_page.field_hs_page_components' => 'node.field_hs_page_components',
+    'paragraph.hs_collection.field_hs_collection_items' => 'paragraph.field_hs_collection_items',
+    'paragraph.hs_row.field_hs_row_components' => 'paragraph.field_hs_row_components',
+    'node.hs_private_page.field_hs_priv_page_components' => 'node.field_hs_priv_page_components',
+  ];
+
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $field_config_storage */
+  $field_config_storage = \Drupal::service('entity_type.manager')->getStorage('field_config');
+
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $fieldConfigStorage */
+  $field_storage = \Drupal::service('entity_type.manager')->getStorage('field_storage_config');
+
+  foreach ($configs_to_change as $field_config => $field_storage_config) {
+
+    // If storage does not yet exist, create it.
+    if (empty($field_storage->load($field_storage_config))) {
+      $field_storage->createFromStorageRecord($config_storage->read('field.storage.' . $field_storage_config))->save();
+    }
+
+    // Update the field instances from filesystem.
+    $config = $field_config_storage->load($field_config);
+    if (empty($config)) {
+      $field_config_storage->create($config_storage->read('field.field.' . $field_config))->save();
+    }
+  }
+
+  // Remove webform existing config - needs to be done before uninstall.
+  $query = \Drupal::database()
+    ->select('config', 'c')
+    ->fields('c', ['name'])
+    ->condition('name', '%' . Database::getConnection()->escapeLike('webform') . '%', 'LIKE')
+    ->execute();
+  $config_names = $query->fetchCol();
+
+  $config_factory = \Drupal::configFactory();
+
+  foreach ($config_names as $config) {
+    $config_factory->getEditable($config)->delete();
+    $message = t('Deleted @config', [
+      '@config' => $config,
+    ]);
+    Drush::logger()->notice($message);
+  }
+
+  // Uninstalls modules after removing configs to prevent updb errors.
+  \Drupal::service('module_installer')->uninstall(['webform_ui', 'webform']);
```

Comment:
Shouldn't `hs_webform` module also be uninstalled here?

Reply:
It correctly gets uninstalled via the `core.extension` but the others needed to be done like this before the config gets imported. So, I can add here, but it's not necessary.